vochain/scrutinizer/scrutinizer.go

```diff
@@ -97,33 +98,109 @@ func NewScrutinizer(dbPath string, app *vochain.BaseApplication) (*Scrutinizer,
 		return nil, err
 	}
 	startTime := time.Now()
-	envelopes, err := s.db.Count(&indexertypes.VoteReference{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total envelopes: %w", err)
-	}
-	entities, err := s.db.Count(&indexertypes.Entity{}, &badgerhold.Query{})
-	if err != nil {
-		return nil, fmt.Errorf("could not count the total entities: %w", err)
-	}
-	processes, err := s.db.Count(&indexertypes.Process{}, &badgerhold.Query{})
+
+	txCountStore, envelopeCountStore, processCountStore, entityCountStore, err := s.retrieveCounts()
 	if err != nil {
-		return nil, fmt.Errorf("could not count the total processes: %w", err)
+		return nil, fmt.Errorf("could not create scrutinizer: %v", err)
 	}
 
-	log.Infof("indexer initialization took %s, stored %d envelopes and %d entities",
+	log.Infof("indexer initialization took %s, stored %d "+
+		"transactions, %d envelopes, %d processes and %d entities",
 		time.Since(startTime),
-		envelopes,
-		entities)
-	s.countTotalEnvelopes = uint64(envelopes)
-	s.countTotalEntities = int64(entities)
-	s.countTotalProcesses = int64(processes)
+		txCountStore.Count,
+		envelopeCountStore.Count,
+		processCountStore.Count,
+		entityCountStore.Count)
 	// Subscrive to events
 	s.App.State.AddEventListener(s)
 	s.envelopeHeightCache = lru.New(countEnvelopeCacheSize)
 	s.resultsCache = lru.New(resultsCacheSize)
 	return s, nil
 }
 
+// retrieveCounts returns a count for txs, envelopes, processes, and entities, in that order.
+// If no CountStore model is stored for the type, it counts all db entries of that type.
+func (s *Scrutinizer) retrieveCounts() (*indexertypes.CountStore,
+	*indexertypes.CountStore, *indexertypes.CountStore, *indexertypes.CountStore, error) {
+	var err error
+	txCountStore := new(indexertypes.CountStore)
+	if err = s.db.Get(indexertypes.CountStoreTransactions, txCountStore); err != nil {
+		log.Warnf("could not get the transaction count: %v", err)
+		count, err := s.db.Count(&indexertypes.TxReference{}, &badgerhold.Query{})
+		if err != nil {
+			if err != badger.ErrKeyNotFound {
+				return nil, nil, nil, nil, fmt.Errorf("could not count total transactions: %v", err)
+			}
+			// If keyNotFound error, ensure count is 0
+			count = 0
+		}
+		// Store new countStore value
+		txCountStore.Count = uint64(count)
+		txCountStore.Type = indexertypes.CountStoreTransactions
+		if err := s.db.Upsert(txCountStore.Type, txCountStore); err != nil {
+			return nil, nil, nil, nil, fmt.Errorf("could not store transaction count: %v", err)
+		}
+	}
+	envelopeCountStore := new(indexertypes.CountStore)
+	if err = s.db.Get(indexertypes.CountStoreEnvelopes, envelopeCountStore); err != nil {
+		log.Warnf("could not get the envelope count: %v", err)
+		count, err := s.db.Count(&indexertypes.VoteReference{}, &badgerhold.Query{})
+		if err != nil {
```

Comment:
count is 0 in case of error (zero-value for uint64) so you can simplify this logic:

```if err != nil &&  err != badger.ErrKeyNotFound { return .... }```

Reply:
Same on all cases.